.github/workflows/build.yaml

```diff
@@ -44,12 +44,38 @@ jobs:
         run: |
           export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
           make test
-        shell: bash
-
-      - name: Login to Docker Hub
-        if: github.ref == 'refs/heads/master'
-        run: echo "${{ secrets.DOCKERHUB_PASSWORD }}" | docker login docker.io --username "${{ secrets.DOCKERHUB_USERNAME }}" --password-stdin
-
-      - name: Build and push image
-        if: github.ref == 'refs/heads/master'
-        run: make docker-build docker-push
\ No newline at end of file
+  push_docker_image:
+    runs-on: ubuntu-latest
+    needs: testing
+    if: github.ref == 'refs/heads/master'
+    name: Build medusa-operator image
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+      - name: Set git parsed values
+        id: vars
+        run: |
+          echo ::set-output name=sha_short::$(git rev-parse --short=8 ${{ github.sha }})
+          echo ::set-output name=tag_name::${GITHUB_REF#refs/tags/}
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: k8ssandra/medusa-operator:${{ steps.vars.outputs.sha_short }}
```

Comment:
Do we want the Docker image tag to be the SHA and not the TAG?

Reply:
This is workflow is just for PRs. `release.yaml` handles tags.